src/EventListener/CsrfTokenCookieListener.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Csrf\MemoryTokenStorage;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Retrieves and stores CSRF token cookies.
+ *
+ * @author Martin Auswöger <martin@auswoeger.com>
+ */
+class CsrfTokenCookieListener
+{
+    /**
+     * @var MemoryTokenStorage
+     */
+    private $tokenStorage;
+
+    /**
+     * @var int
+     */
+    private $cookieLifetime;
+
+    /**
+     * @var string
+     */
+    private $cookiePrefix;
+
+    /**
+     * Constructor.
+     *
+     * @param MemoryTokenStorage $tokenStorage
+     * @param int                $cookieLifetime
+     * @param string             $cookiePrefix
+     */
+    public function __construct(MemoryTokenStorage $tokenStorage, int $cookieLifetime = 86400, string $cookiePrefix = 'csrf_')
```

Comment:
Does this have to be a `MemoryTokenStorage`? Shouldn't we expect the interface?

Reply:
We call the `initialize()` method on it, so we cannot use the interface.